### PR TITLE
Create .spi.yml

### PR DESCRIPTION
**Pull request by finestructure**
As discussed with @franklinsch , this will ensure SPI documentation is updated without any particular workarounds in place in SPI.

We initially added custom support to host documentation on SPI without a `.spi.yml` file but this is becoming a bit tricky to maintain and has broken in a couple of instances.

### CODE REVIEW

File: .spi.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+version: 1
+builder:
+  configs:
+    - documentation_targets: [SwiftDocCPlugin]
```
Comment (finestructure) [inline .spi.yml:4]: There are a couple of other targets in the package but I believe this is the one we'd configured in our override. Or should there be others in this list as well?

Comment (franklinsch) [reply .spi.yml:4]: @ethan-kusters can you confirm?

Comment (ethan-kusters) [reply .spi.yml:4]: That's correct 👍🏻

**Review by ethan-kusters — APPROVED** [review]

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Pull request merged**